apps/mull-api/src/app/auth/auth.service.ts

```diff
@@ -1,49 +1,67 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException, UnauthorizedException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
 import { compare } from 'bcrypt';
-import { Request } from 'express';
+import { Request, Response } from 'express';
 
+import { environment } from '../../environments/environment';
+import { RegistrationMethod } from '@mull/types';
 import { User } from '../entities';
 import { CreateUserInput } from '../user/inputs/user.input';
 import { UserService } from '../user/user.service';
 
 @Injectable()
 export class AuthService {
-  constructor(private userService: UserService) {}
+  constructor(private userService: UserService, private jwtService: JwtService) {}
 
-  authLogin(req: Request) {
-    if (!req.user) return null;
-    return { user: req.user };
+  authLogin({ user }: Request, res: Response) {
+    if (!user) return null;
+    const accessToken = this.createAccessToken(user);
+    this.sendRefreshToken(res, this.createRefreshToken(user));
+    res.redirect(`${environment.client.baseUrl}/token-redirect/${accessToken}`);
   }
 
   async validateOAuthUser(
     user: Partial<User>,
     callback: (error: string, user?: Partial<User>, info?: string) => void
   ) {
-    const users: User[] = await this.userService.findByEmail(user.email);
-    let hasSameRegistrationMethod = false;
-    if (users.length >= 1) {
-      users.forEach(({ registrationMethod }: Partial<User>, idx) => {
-        if (user.registrationMethod === registrationMethod) {
-          hasSameRegistrationMethod = true;
-          callback(null, users[idx]);
-        }
-      });
-    }
-    if (!hasSameRegistrationMethod) {
+    const users: User[] = await this.userService.findUnique(user.email, user.registrationMethod);
+    if (users.length == 1) callback(null, users[0]);
+    else {
       const newUser = await this.userService.create(user as CreateUserInput);
       callback(null, newUser);
     }
   }
 
+  createAccessToken(user: Partial<User>) {
+    return this.jwtService.sign(
+      { id: user.id },
+      { expiresIn: '15m', secret: environment.jwt.accessSecret }
+    );
+  }
+
+  createRefreshToken(user: Partial<User>) {
+    return this.jwtService.sign(
+      { id: user.id, tokenVersion: user.tokenVersion },
+      { expiresIn: '7d', secret: environment.jwt.refreshSecret }
+    );
+  }
+
+  sendRefreshToken(res: Response, refreshToken: string) {
+    res.cookie('mullToken', refreshToken, {
+      maxAge: 7 * 24 * 3600 * 1000,
+      secure: environment.production,
```

Comment:
A comment might be helpful so future readers don't have to go through documentation 
```suggestion
      secure: environment.production, // determines if cookie is to be used with HTTPS only 
```